### PR TITLE
Updated parser and transformation rules for new asciimath examples

### PR DESCRIPTION
**Pull request by suleman-uzair**
Updated AsciiMath parser and transformation rules for new examples from **```asciidoctor/asciimath```**.

### CODE REVIEW

**Comment by ronaldtse** [comment]
@suleman-uzair can we merge this or is this still a draft? Also could you help rebase to fix the conflicts?

**Comment by suleman-uzair** [comment]
> @suleman-uzair can we merge this or is this still a draft? Also could you help rebase to fix the conflicts?

@ronaldtse This PR is yet in progress.

File: spec/plurimath/math/formula_spec.rb [hunk 1]
```diff
@@ -174,14 +174,14 @@
     context "contains log formula string with text" do
       let(:exp) { "log(1+2+3+4)^(\"4 terms\")" }
       it 'converts formula back to Asciimath string for log function' do
-        expect(formula).to eql('log_(1+2+3+4)^"4 terms"')
+        expect(formula).to eql('log1+2+3+4^"4 terms"')
```
Comment (ronaldtse) [inline spec/plurimath/math/formula_spec.rb:177]: This change doesn't seem correct.

The original:
```ruby
"log(1+2+3+4)^(\"4 terms\")"
```

And the converted back formula:
```ruby
'log1+2+3+4^"4 terms"'
```

Are rather different in element precedence.

File: spec/plurimath/math/formula_spec.rb [hunk 2]
```diff
@@ -174,14 +174,14 @@
...
       end
     end
 
     context "contains oint formula string with text" do
       let(:exp) { "oint(1+2+3+4)^(text(4 terms))" }
       it 'converts formula back to Asciimath string for oint function' do
-        expect(formula).to eql('oint_(1+2+3+4)^"4 terms"')
+        expect(formula).to eql('oint1+2+3+4^"4 terms"')
```
Comment (ronaldtse) [inline spec/plurimath/math/formula_spec.rb:184]: Doesn't seem right. Should have been `oint(1+2+3+4)^"4 terms"`

File: spec/plurimath/math/formula_spec.rb [hunk 3]
```diff
@@ -223,7 +223,7 @@
     context "contains underset formula string" do
       let(:exp) { "underset(sum(sin(theta)))(i)" }
       it 'converts formula back to Asciimath string for underset function' do
-        expect(formula).to eql('underset(sum_(sin(theta)))(i)')
+        expect(formula).to eql('underset(sumsin(theta))(i)')
```
Comment (ronaldtse) [inline spec/plurimath/math/formula_spec.rb:226]: Doesn't seem right. Should have been `underset(sum(sin(theta)))(i)`.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thank you @suleman-uzair !

**Comment by ronaldtse** [comment]
> Updated AsciiMath parser and transformation rules for new examples from asciidoctor/asciimath.

@suleman-uzair can you help mark which specs come from asciimath so we can track them if they change in the future? Thanks!

**Pull request merged**